src/lib.rs

```diff
@@ -475,8 +545,23 @@ impl<T> ops::DerefMut for Ref<'_, T> {
 impl<T> Drop for Ref<'_, T> {
     #[inline]
     fn drop(&mut self) {
-        test_println!("drop Ref<{}>", core::any::type_name::<T>());
-        test_dbg!(self.slot.state.store(test_dbg!(self.new_state), Release));
+        if self.is_pop {
+            test_println!("drop Ref<{}> (pop)", core::any::type_name::<T>());
+            test_dbg!(self
+                .slot
+                .state
+                .fetch_update(SeqCst, SeqCst, |state| {
+                    Some(test_dbg!(clear_has_reader(state)))
+                })
+                .unwrap_or_else(|_| unreachable!()));
```

Comment:
can't this just be a `fetch_and` operation to clear the bit, rather than `fetch_update` (which performs a CAS under the hood):
```suggestion
            test_dbg!(self
                .slot
                .state
                .fetch_and(!(1 << (usize::BITS - 1)), SeqCst))
```
(this might look a bit nicer if we had a constant for `1 << (usize::BITS - 1))`...)